DependencyInjection/Compiler/PointcutMatchingPass.php

```diff
@@ -99,7 +99,11 @@ private function processDefinition(Definition $definition, $pointcuts, &$interce
             return;
         }
 
-        if ($definition->getFactoryService() || $definition->getFactoryClass()) {
+        // Symfony 2.6 getFactory method
+        // TODO: Use only getFactory when bumping require to Symfony >= 2.6
+        if (method_exists($definition, 'getFactory') && $definition->getFactory()) {
+            return;
+        } elseif (!method_exists($definition, 'getFactory') && ($definition->getFactoryService() || $definition->getFactoryClass())) {
```

Comment:
IMO, since in previous block there is a return statement and this condition must be removed when bumping to Symfony 2.6+, is better to move to a "standalone" ```if``` instead ```elseif```.